sssd_test_framework/roles/ipa.py

```diff
@@ -2194,3 +2232,721 @@ def delete(self) -> None:
+        return self
+
+    def modify(
+            self,
+            *,
+            description: str | None = None,
+    ) -> IPAHbacServiceGroup:
+        """
+        Modify existing IPA HBAC service group.
+
+        :param description: Description, defaults to None
+        :type description: str | None, optional
+        :return: Self.
+        :rtype: IPAHBACServiceGroup
+        """
+        attrs = {
+            "desc": (self.cli.option.VALUE, description),
+        }
+
+        self._modify(attrs)
+        return self
+
+    def add_member(
+            self,
+            *,
+            hbacsvc: list[str] | str | None = None,
+            hbacsvcgroup: list[str] | str | None = None,
+    ) -> IPAHbacServiceGroup:
+        """
+        Add HBAC service group members.
+
+        :param hbacsvc: HBAC service(s) to add as member(s).
+        :type hbacsvc: list[str] | str | None, optional
+        :param hbacsvcgroup: HBAC service group(s) to add as member(s).
+        :type hbacsvcgroup: list[str] | str | None, optional
+        :return: Self.
+        :rtype: IPAHBACServiceGroup
+        """
+        cmd = ["ipa", f"{self.command_group}-add-member", self.name]
+
+        if hbacsvc:
+            services_list = [hbacsvc] if isinstance(hbacsvc, str) else hbacsvc
+            for service in services_list:
+                cmd.append(f"--hbacsvcs={service}")
+
+        if hbacsvcgroup:
+            servicegroups_list = [hbacsvcgroup] if isinstance(hbacsvcgroup, str) else hbacsvcgroup
+            for servicegroup in servicegroups_list:
+                cmd.append(f"--hbacsvcgroups={servicegroup}")
+
+        if hbacsvc or hbacsvcgroup:
+            self.role.host.conn.exec(cmd)
+
+        return self
+
+    def remove_member(
+            self,
+            *,
+            hbacsvc: list[str] | str | None = None,
+            hbacsvcgroup: list[str] | str | None = None,
+    ) -> IPAHbacServiceGroup:
+        """
+        Remove HBAC service group members.
+
+        :param hbacsvc: HBAC service(s) to remove as member(s).
+        :type hbacsvc: list[str] | str | None, optional
+        :param hbacsvcgroup: HBAC service group(s) to remove as member(s).
+        :type hbacsvcgroup: list[str] | str | None, optional
+        :return: Self.
+        :rtype: IPAHBACServiceGroup
+        """
+        cmd = ["ipa", f"{self.command_group}-remove-member", self.name]
+
+        if hbacsvc:
+            services_list = [hbacsvc] if isinstance(hbacsvc, str) else hbacsvc
+            for service in services_list:
+                cmd.append(f"--hbacsvcs={service}")
+
+        if hbacsvcgroup:
+            servicegroups_list = [hbacsvcgroup] if isinstance(hbacsvcgroup, str) else hbacsvcgroup
+git            for servicegroup in servicegroups_list:
+                cmd.append(f"--hbacsvcgroups={servicegroup}")
+
+        if hbacsvc or hbacsvcgroup:
+            self.role.host.conn.exec(cmd)
+
+        return self
+
+
+class IPAHbac(IPAObject):
```

Comment:
This is a lot for a class. Is there a reason why you didn't parameterized one add method, than add_service, add_host, add_user_group?